sys/phydat/phydat_str.c

```diff
@@ -78,14 +80,35 @@ const char *phydat_unit_to_str(uint8_t unit)
     }
 }
 
-char phydat_scale_to_str(int8_t scale)
+char phydat_scale_to_str(int8_t scale, uint8_t unit)
 {
+    /* extra handling for multi and non dimensional units */
+    switch (unit) {
+        case UNIT_M2:       /* rescale for 2d unit */
+            if (scale % 2 != 0) {
+                return '\0';
+            }
+            scale /= 2;
+            break;
+        case UNIT_M3:       /* rescale for 3d unit */
+            if (scale % 3 != 0) {
+                return '\0';
+            }
+            scale /= 3;
+            break;
+        case UNIT_UNDEF:
+        case UNIT_NONE:
+        case UNIT_PERCENT:
```

Comment:
yeah there is bunch of units where scale2str makes no sense, I don't know if we really want to cover all cases, or just hope for sensible users to not scale those units anyway?